### DISCUSSION TITLE
Tesla deleted my Solar-only site and merged it with my Powerwall site

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#248] [General]
It's fortunate that @mcbirse got the Solar-only history script working when he did, because as of about a week ago, my account no longer shows two separate sites in the API. As of right now, I can't pull the solar-only history on my account. Luckily I pulled all the history that was available prior to that happening.

Last week, while away on a vacation, I noticed that my solar production had dropped out mid-day. (When I got back home I found a breaker had tripped between the inverter and the Powerwall Gateway). When I went to the Tesla app to confirm what I was seeing in Powerwall Dashboard, I no longer had the icon/tab in the app for my solar site. I only had the tab for my Powerwall data. Since I have had a Solar City PPA system since 2016, I had that tab in the app prior to getting my Powerwalls in 2022, and since getting the Powerwalls I've had two separate sections in the app.

The big complication is that I have a second solar array going online next Tuesday that I own. The two solar array outputs are combined before being fed to the Powerwall Gateway 2. If Tesla isn't maintaining a separate site for the PPA, how are they tracking the PPA production that I have to pay for separately from the new production that I own?

I'm still emailing back and forth with Tesla Energy support, but my concern now is whether they have combined the two sites for billing purposes. They initially said they hadn't, but since I only have one site showing in the API, I'm not trusting that answer. I also can't pull any data before 2022 in the app. The customer support agent tried to change that so I could see older data from within the Powerwall section, but was unsuccessful.

The only explanation I got for the change was " I think an update with the app features might’ve caused this, we like to have all sites that belong to a single property together for better visibility."

**Comment by mcbirse** [comment]
Yikes!! That's pretty bad of Tesla. Glad to hear you had gotten the history prior to this occurring. 

It seems we can't rely on the data always being available for us from the Tesla cloud. It certainly highlights the importance of those backups yet again as well.

My paranoia about having backups and data retention has been validated, thank you. I will continue my process of multiple backups including offline backups at a remote physical location! 😃

**Reply by jasonacox** [reply]
Agree.  Yikes.
